src/components/Order/useStep.js

```diff
@@ -0,0 +1,25 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const useStep = (totalSteps, finalRoute = "/#TODO") => {
+  const [activeStep, setActiveStep] = useState(0);
+  const navigate = useNavigate();
+
+  const next = () => {
+    if (activeStep === totalSteps - 1) 
+    {
+      navigate(finalRoute);
```

Comment:
1 - next (the name) does not indicate that this is a function that is doing something. 
I would call this handleNext or something that will indicate the user of this hook that this is a function and not a variable. 

2 - you can use guard clauses for early return in your function thus making your function smaller and more readable, it would be something like this : 

```js
const next = () => {
  if (activeStep === totalSteps - 1) {
    navigate(finalRoute);
    return;
  }

  setActiveStep(prev => prev + 1);
};

```